libs/dao/src/main/java/com/akto/dao/testing_run_findings/TestingRunIssuesDao.java

```diff
@@ -1,13 +1,159 @@
+
+        pipeline.add(Aggregates.group(groupedId, Accumulators.sum("count", 1)));
+
+        MongoCursor<BasicDBObject> severitiesCursor = TestingRunIssuesDao.instance.getMCollection().aggregate(pipeline, BasicDBObject.class).cursor();
+        while(severitiesCursor.hasNext()) {
+            try {
+                BasicDBObject basicDBObject = severitiesCursor.next();
+                String severity = ((BasicDBObject) basicDBObject.get("_id")).getString("severity");
+                int apiCollectionId = ((BasicDBObject) basicDBObject.get("_id")).getInt("apiCollectionId");
+                int count = basicDBObject.getInt("count");
+                if(resultMap.containsKey(apiCollectionId)){
+                    Map<String,Integer> severityMap = resultMap.get(apiCollectionId);
+                    severityMap.put(severity, count);
+                }else{
+                    Map<String,Integer> severityMap = new HashMap<>();
+                    severityMap.put(severity, count);
+                    resultMap.put(apiCollectionId, severityMap);
+                }
+
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+        return resultMap;
+    }
+  
+    public Map<String, Integer> getTotalSubcategoriesCountMap(int startTimeStamp, int endTimeStamp){
+        List<Bson> pipeline = new ArrayList<>();
+
+        pipeline.add(Aggregates.match(Filters.and(
+                Filters.eq(TestingRunIssues.TEST_RUN_ISSUES_STATUS, "OPEN"),
+                Filters.lte(TestingRunIssues.LAST_SEEN, endTimeStamp),
+                Filters.gte(TestingRunIssues.LAST_SEEN, startTimeStamp)
+            )
+        ));
+        BasicDBObject groupedId = new BasicDBObject("subCategory", "$_id.testSubCategory");
+        pipeline.add(Aggregates.group(groupedId, Accumulators.sum("count", 1)));
+
+        Map<String,Integer> result = new HashMap<>();
+        MongoCursor<BasicDBObject> severitiesCursor = TestingRunIssuesDao.instance.getMCollection().aggregate(pipeline, BasicDBObject.class).cursor();
+        while(severitiesCursor.hasNext()) {
+            try {
+                BasicDBObject basicDBObject = severitiesCursor.next();
+                String subCategory = ((BasicDBObject) basicDBObject.get("_id")).getString("subCategory");
+                Integer count = basicDBObject.getInt("count");
+                result.put(subCategory, count);
+            }catch(Exception e){
+                e.printStackTrace();
+            }
+        }
+
+        return result;
+    }
+
+    public Map<String, Integer> getTotalSeveritiesCountMap(int startTimeStamp, int endTimeStamp){
+        List<Bson> pipeline = new ArrayList<>();
+
+        pipeline.add(Aggregates.match(Filters.and(
+                Filters.eq(TestingRunIssues.TEST_RUN_ISSUES_STATUS, "OPEN"),
+                Filters.lte(TestingRunIssues.LAST_SEEN, endTimeStamp),
+                Filters.gte(TestingRunIssues.LAST_SEEN, startTimeStamp)
```

Comment:
keep last seen filters in front, because they are indexes

Reply:
status is also an index